pkg/app/app.go

```diff
@@ -243,13 +243,19 @@ func CreateApp() App {
 	metrics.Init(app.log.Logger)
 	app.log.Debug("Creating AnsibleBroker")
 	if app.broker, err = broker.NewAnsibleBroker(
-		app.dao, app.log.Logger, app.config.Openshift, app.registry, *app.engine, app.config.Broker,
+		app.dao, app.dao, app.log.Logger, app.config.Openshift, app.registry, *app.engine, app.config.Broker,
```

Comment:
I can see the dao ending up with a bunch of interfaces, which could blow up this construction method.

Reply:
Yes agree. Something that may help here, would be to split the broker into the different actions.

In sudo code. Something like :

```go
package broker

type provision struct{}

func NewProvision(<dao interface>, <dao interface>, logger)*provision{}

func (p *provision)Handle(req, ProvisionRequest, ...)(ProvisionResponse,error){}

```
could work?

Reply:
I could try this approach for the last operation handler to aid discussion.